test/unit/document_test.rb

```diff
@@ -267,4 +267,31 @@ class DocumentTest < ActiveSupport::TestCase
       document.save
     end
   end
+
+  test "#withdrawals returns withdrawals that have happened on editions of the document" do
+    # Create & publish a document
+    document = create(:document)
+    edition = create(:edition, document: document)
+    user = edition.creator
+    EditionForcePublisher.new(edition).perform!
+
+    3.times do
```

Comment:
Not necessarily a blocker, but this is quite an expensive unit test. Can we just create the objects in the desired state?

Reply:
Yeah – fair point! I wasn't sure which approach was best.

I'll take a look at both approaches and see what the performance impact is.

Reply:
So, in terms of performance, the test takes:

- **~1.5 seconds** to run when using the `EditionForcePublisher`, `EditionWithdrawer` and `EditionUnwithdrawer` services
- **~0.2 seconds** to run when creating objects directly

Quite a difference! 😅

I think your point about the test feeling more integration(ish) is completely valid. My initial concern was that just creating objects directly doesn't realistically reproduce changes that'd happen in related models (e.g. older editions becoming "superseded" rather than remaining "withdrawn"). So I wanted to setup a more realistic state.

But you're absolutely right. Since this is a unit test, the other side effects aren't really as important. We can setup everything this method needs without the overhead of using those higher level services.

Reply:
I've just pushed up that change. Thanks for suggesting it @davidgisbey 🤓